### PR TITLE
Use a semaphore to trigger shutdown procedures

### PR DESCRIPTION
**Pull request by skeees**
Very simple cleanup to avoid sleep waiting on the `fShutdownRequested` flag

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -131,16 +131,26 @@ static const char* FEE_ESTIMATES_FILENAME="fee_estimates.dat";
 //
 
 std::atomic<bool> fRequestShutdown(false);
+std::atomic<bool> fDumpMempoolLater(false);
```
Comment (MarcoFalke) [inline src/init.cpp]: Unused and unrelated?

Comment (skeees) [reply src/init.cpp]: must be a stray rebase - will fix

**Comment by sipa** [comment]
I think this breaks SIGINT/SIGTERM handling.

**Comment by skeees** [comment]
Yes - you're right :( let me see if there's an easy safe way to do that, but might be more trouble than its worth.

**Comment by sipa** [comment]
You can use a condition variable and have a loop with short-timeout waits on it. The signal handlers can't do anything but set variables, and will need a wait loop, but everything else can use condition variable waking.

**Comment by laanwj** [comment]
You can't use mutexes inside signal handlers.
What would work is to create a pipe.

See also discussion in #10873.

**Comment by laanwj** [comment]
Closing in favor of  #13211.

**Pull request closed without merging**